src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GrapplingHookListener.java

```diff
@@ -136,6 +139,23 @@ public void onPortalEnter(EntityPortalEnterEvent e) {
         }
     }
 
+    // Fixing Issue #2351
+    @EventHandler
+    public void onLeash(PlayerLeashEntityEvent e) {
+        if (!isEnabled()) {
+            return;
+        }
+
+        Player p = (Player)e.getPlayer();
```

Comment:
No need to cast a Player to a Player here.

Reply:
Oh right Sorry! It was the force of habbit from commands :) I will fix that and submit!